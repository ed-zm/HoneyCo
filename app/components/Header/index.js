import React from 'react'
import styled from 'styled-components'
import { HoneyCoBlueRGBA, HoneyCoYellow, HoneyCoDarkGray } from '../../constants'

const StyledJumbotron = styled.div`
  background-image: url(${(props) => props.background});
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  bakground-color: ${HoneyCoBlueRGBA(0.5)};
  width: 100%;
  height: 650px;
  position: relative;
`
const StyledJumbotronTransparency = styled.div`
  background-color: rgba(77, 162, 159, 0.6);
  width: 100%;
  height: 650px;
  display: flex;
  align-items: center;
  justify-content: center;
`

const JumbotronContainer = styled.div`
  padding: 1rem;
  color: white;
  text-weight: 700;
  
  // MAKE CHANGES BELOW TO PUT TEXT/BUTTONS ON TOP OF THE JUMBOTRON
  h1 {
    font-size: 1.5rem;
  }
  
  @media (min-width: 576px) {
    h1 {
      font-size: 1.5rem;
    }
  }
  @media (min-width: 768px) {
    h1 {
      font-size: 2rem;
    }
  }
  @media (min-width: 992px) {
    h1 {
      font-size: 2.5rem;
    }
  }
  @media (min-width: 1200px) {
    h1 {
      font-size: 4rem;
    }
`

const StyledTextJumbotron = styled.span`
  display: block;
  font-size: 80px;
  font-weight: 500;
  text-align: center;
  @media (max-width: 576px) {
    font-size: 30px;
  }
@media (max-width: 768px) {
    font-size: 40px;
  }
@media (max-width: 992px) {
    font-size: 70px;
  }
@media (max-width: 1200px) {
    font-size: 80px;
  }
`
const StyledSubTextJumbotron = styled.span`
  display: block;
  font-size: 32px;
  margin: 32px auto;
  text-align: center;
`

const StyledButtonJumbotron = styled.button`
  font-family: Oswald;
  font-size: 32px;
  color: ${HoneyCoDarkGray};
  background-color: ${HoneyCoYellow};
  padding: .2em;
  border-radius: 5px;
  border: none;
  padding: .3em 1.5em;
  text-align: center;
  margin-top: 10px;
`

const StyledButtonJumbotronContainer = styled.div`
  text-align: center;
`

const Header = (props) =>
  <StyledJumbotron background = {props.background}>
    <StyledJumbotronTransparency>
      <JumbotronContainer w="topLeft">
        <StyledTextJumbotron>{props.headerText.firstText}</StyledTextJumbotron>
        <StyledTextJumbotron>{props.headerText.secondText}</StyledTextJumbotron>
        <StyledSubTextJumbotron>{props.headerText.subText}</StyledSubTextJumbotron>
        <StyledButtonJumbotronContainer>
          <StyledButtonJumbotron>
            {props.headerText.button}
          </StyledButtonJumbotron>
        </StyledButtonJumbotronContainer>
      </JumbotronContainer>
    </StyledJumbotronTransparency>
  </StyledJumbotron>

export default Header
