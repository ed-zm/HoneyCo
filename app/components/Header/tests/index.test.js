import React from 'react'
import { shallow } from 'enzyme'
import Header from '../index'

const headerText = {
  firstText: 'First Text',
  secondText: 'Second Text',
  subText: 'Sub text',
  button: 'Button',
}

const background = '/someurl'

describe('<Header />', () => {
  it('It should render the Header Component', () => {
    const wrapper = shallow(<Header background = {background} headerText = {headerText}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
