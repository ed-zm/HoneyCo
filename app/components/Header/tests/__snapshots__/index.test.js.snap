// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> It should render the Header Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div
    background="/someurl"
>
    <styled.div>
        <styled.div
            w="topLeft"
        >
            <styled.span>
                First Text
            </styled.span>
            <styled.span>
                Second Text
            </styled.span>
            <styled.span>
                Sub text
            </styled.span>
            <styled.div>
                <styled.button>
                    Button
                </styled.button>
            </styled.div>
        </styled.div>
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div
      background="/someurl"
>
      <styled.div>
            <styled.div
                  w="topLeft"
            >
                  <styled.span>
                        First Text
                  </styled.span>
                  <styled.span>
                        Second Text
                  </styled.span>
                  <styled.span>
                        Sub text
                  </styled.span>
                  <styled.div>
                        <styled.button>
                              Button
                        </styled.button>
                  </styled.div>
            </styled.div>
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Header
        background="/someurl"
        headerText={
                Object {
                        "button": "Button",
                        "firstText": "First Text",
                        "secondText": "Second Text",
                        "subText": "Sub text",
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "background": "/someurl",
          "headerText": Object {
            "button": "Button",
            "firstText": "First Text",
            "secondText": "Second Text",
            "subText": "Sub text",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div
          background="/someurl"
>
          <styled.div>
                    <styled.div
                              w="topLeft"
                    >
                              <styled.span>
                                        First Text
                              </styled.span>
                              <styled.span>
                                        Second Text
                              </styled.span>
                              <styled.span>
                                        Sub text
                              </styled.span>
                              <styled.div>
                                        <styled.button>
                                                  Button
                                        </styled.button>
                              </styled.div>
                    </styled.div>
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div
          background="/someurl"
>
          <styled.div>
                    <styled.div
                              w="topLeft"
                    >
                              <styled.span>
                                        First Text
                              </styled.span>
                              <styled.span>
                                        Second Text
                              </styled.span>
                              <styled.span>
                                        Sub text
                              </styled.span>
                              <styled.div>
                                        <styled.button>
                                                  Button
                                        </styled.button>
                              </styled.div>
                    </styled.div>
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header
    background="/someurl"
    headerText={
        Object {
            "button": "Button",
            "firstText": "First Text",
            "secondText": "Second Text",
            "subText": "Sub text",
          }
    }
/>,
}
`;
